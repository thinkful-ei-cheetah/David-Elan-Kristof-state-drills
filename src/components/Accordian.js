import React from 'react'


class Accordian extends React.Component {

  state = {
    showSection: null
  }

  sections = [
    {
      title: 'Section 1',
      content: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.',
    },
    {
      title: 'Section 2',
      content: 'Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!',
    },
    {
      title: 'Section 3',
      content: 'Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?',
    },
  ]

  selectedSectionName = () => {
    return (
      <li>
        <p>{this.sections[this.state.showSection].content}</p>
      </li>
    )
  }

  handleClick=(sectionNumber)=>{
    this.setState({
      showSection: sectionNumber
    })
  }

  render() {
    return(
      <React.Fragment>
        <div>
          <button onClick={() => this.handleClick(0)}>Section 1</button>
          <button onClick={() => this.handleClick(1)}>Section 2</button>
          <button onClick={() => this.handleClick(2)}>Section 3</button>
        </div>
        <ul>
          {this.state.showSection !== null && this.selectedSectionName(this.state.showSection)}
        </ul>
        </React.Fragment>  
    )
  }
} 


export default Accordian;