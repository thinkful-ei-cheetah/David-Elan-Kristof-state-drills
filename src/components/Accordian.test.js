import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Accordian from './Accordian';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

describe('Accordian works!', ()=>{
  // renders an empty li when sections prop is not supplied
  it('Renders without crashing', ()=>{
    const div = document.createElement('div');
    ReactDOM.render(<Accordian />, div )
    ReactDOM.unmountComponentAtNode(div)
    });
    // renders no sections as active by default
  it('Renders to the page with no selection', ()=>{
  const tree = renderer
    .create(<Accordian />)
    .toJSON();
    expect(tree).toMatchSnapshot();  
  })
  // opens a clicked section
  it('opens a clicked session', () => {
    const wrapper = shallow(<Accordian />)
    wrapper.find('button').at(1).simulate('click')
    expect(toJson(wrapper)).toMatchSnapshot()
  })
  it('only opens one section at a time', () => {
    const wrapper = shallow(<Accordian />)
    wrapper.find('button').at(1).simulate('click')
    wrapper.find('button').at(2).simulate('click')
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})

