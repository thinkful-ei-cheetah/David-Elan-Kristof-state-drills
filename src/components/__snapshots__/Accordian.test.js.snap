// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordian works! Renders to the page with no selection 1`] = `
Array [
  <div>
    <button
      onClick={[Function]}
    >
      Section 1
    </button>
    <button
      onClick={[Function]}
    >
      Section 2
    </button>
    <button
      onClick={[Function]}
    >
      Section 3
    </button>
  </div>,
  <ul />,
]
`;

exports[`Accordian works! only opens one section at a time 1`] = `
<Fragment>
  <div>
    <button>
      Section 1
    </button>
    <button>
      Section 2
    </button>
    <button>
      Section 3
    </button>
  </div>
  <ul />
</Fragment>
`;

exports[`Accordian works! opens a clicked session 1`] = `
<Fragment>
  <div>
    <button>
      Section 1
    </button>
    <button>
      Section 2
    </button>
    <button>
      Section 3
    </button>
  </div>
  <ul />
</Fragment>
`;
